src/sshRunner.ts

```diff
@@ -99,48 +111,78 @@ export class SSHRunner extends TerminalBaseRunner {
-                            if (utilities.isTelemetryEnabled()) {
-                                terminal.sendText('export ' + Constants.UserAgentName + '=' + utilities.getUserAgent());
+            }
+        }
+
+        if (terminal) {
+            terminal.show();
+
+            cmds.push('ansible-playbook ' + targetPlaybook);
+            this.sendCommandsToTerminal(terminal, cmds);
+
+        } else {
+            openSSHConsole(this._outputChannel, targetServer)
+                .then((term) => {
+                    if (!term) {
+                        this._outputChannel.appendLine('\nSSH connection failed.');
+                        this._outputChannel.show();
+                        return;
+                    }
+                    this.terminalList[this.getTerminalName(targetServer.host)] = term;
+
+                    var count: number = 60;
+                    var _localthis = this;
+                    var connected = false;
+
+                    const tempFile = path.join(os.tmpdir(), 'vscodeansible-ssh-' + targetServer.host + '.log');
+
+                    var interval = setInterval(function () {
+                        count--;
+                        if (count > 0) {
+                            if (fs.existsSync(tempFile)) {
+                                count = 0;
+                                fs.removeSync(tempFile);
+                                connected = true;
+
+                                cmds.push('ansible-playbook ' + targetPlaybook);
+                                _localthis.sendCommandsToTerminal(term, cmds);
+                                term.show();
                             }
+                        } else {
+                            clearInterval(interval);
 
-                            for (let cmd of cmds) {
-                                terminal.sendText(cmd);
+                            if (!connected) {
+                                _localthis._outputChannel.appendLine('\nFailed to connect to ' + targetServer.host + ' after 30 seconds');
                             }
-                            terminal.sendText('ansible-playbook ' + targetPlaybook);
-                            terminal.show();
                         }
-                    } else {
-                        clearInterval(interval);
+                    }, 500);
+                });
+        }
+    }
 
-                        if (!connected) {
-                            _localthis._outputChannel.appendLine('\nFailed to connect to ' + targetServer.host + ' after 30 seconds');
-                        }
-                    }
-                }, 500);
-            });
+    private getTerminalName(host: string): string {
+        return 'SSH ' + host;
+    }
 
+    private sendCommandsToTerminal(terminal: vscode.Terminal, cmds: string[]): void {
+        if (utilities.isTelemetryEnabled()) {
+            terminal.sendText('export ' + Constants.UserAgentName + '=' + utilities.getUserAgent());
```

Comment:
so this will be sent to terminal even if terminal is reused and these variables are already set?

Reply:
actually i think there should be if (telemetry) sendCommands... (telemetry commands) when terminal is created.

Reply:
those variables may change.